extensions/notebook/src/book/bookModel.ts

```diff
@@ -186,12 +186,13 @@ export class BookModel {
 				const config = yaml.safeLoad(fileContents.toString());
 				fileContents = await fsPromises.readFile(this._tableOfContentsPath, 'utf-8');
 				let tableOfContents: any = yaml.safeLoad(fileContents.toString());
+				const parsedTOC: IJupyterBookToc = { sections: this.parseJupyterSections(this._bookVersion, tableOfContents) };
 				let book: BookTreeItem = new BookTreeItem({
 					version: this._bookVersion,
 					title: config.title,
 					contentPath: this._tableOfContentsPath,
 					root: this.bookPath,
-					tableOfContents: { sections: this.parseJupyterSections(this._bookVersion, tableOfContents) },
+					tableOfContents: parsedTOC,
```

Comment:
So we're going to be sending across the entire toc now with the drag and drop stuff? Any concerns about the size of that causing issues? We still may want to consider having the  actual book item we give to VS Code be an object with only a subset of the properties if we actually need the table of contents stored here

Reply:
Chatted offline - room for improvement here since TreeItems shouldn't really know about other tree items or the model as a whole. Here's how I'm thinking we should have it set up 

BookTreeItemProvider -> BookModel -> BookTreeItems

So the provider has a list of BookModels that each contain the full book structure. Those book models have a single top-level TreeItem for the root of the book, and then each section and file in the structure has its own BookTreeItem (which are all kept in some sort of structure internal to the BookModel). Each BookTreeItem then only needs to contain the information necessary to display it and an ID to do lookups with - the provider/model will have mapping tables for taking the tree items/IDs and getting the actual section/file that they correspond to.